### PR TITLE
ref(test): Convert IssueListSearchBar test to RTL

### PR DESCRIPTION
**Pull request by vuluongj20**
Another one ✨

### CODE REVIEW

File: static/app/views/issueList/searchBar.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+              isOrgCustom: false,
+              isGlobal: false,
+              type: 0,
+            }}
+          />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByLabelText('Unpin this search'));
```
Comment (evanpurkhiser) [inline static/app/views/issueList/searchBar.spec.tsx:290]: `getByRole('button', {name: 'Unpin this search'})`?

File: static/app/views/issueList/searchBar.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+    it('pins a search from the searchbar', function () {
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} query='url:"fu"' />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByTestId('pin-icon'));
```
Comment (evanpurkhiser) [inline static/app/views/issueList/searchBar.spec.tsx:253]: Should be able to use `getByRole` here also if it's labeled right

Comment (vuluongj20) [reply static/app/views/issueList/searchBar.spec.tsx]: Ahh true! https://github.com/getsentry/sentry/pull/40289/commits/12bb1baa669ec644f5b12d41d67bc6981af699b2

File: static/app/views/issueList/searchBar.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} tagValueLoader={loader} />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByRole('textbox'));
+      userEvent.keyboard('timesSeen:');
```
Comment (evanpurkhiser) [inline static/app/views/issueList/searchBar.spec.tsx:99]: oh I've been using `type`

File: static/app/views/issueList/searchBar.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+            {...defaultProps}
+            tagValueLoader={loader}
+            onSearch={onSearch}
+          />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByRole('textbox'));
+      userEvent.keyboard('url:"fu"');
```
Comment (evanpurkhiser) [inline static/app/views/issueList/searchBar.spec.tsx:127]: you can do
```suggestion
      userEvent.type(screen.getByRole('textbox'), 'url:"fu"');
```

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/views/issueList/searchBar.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+      );
+
+      const textarea = screen.getByRole('textbox');
+
+      // Keyboard navigate to first item and select
+      userEvent.click(textarea);
+      userEvent.keyboard('t');
+      await waitFor(() =>
+        expect(screen.getAllByTestId('search-autocomplete-item')[0]).toBeInTheDocument()
+      );
```
Comment (evanpurkhiser) [inline static/app/views/issueList/searchBar.spec.tsx:186]:
```suggestion
      expect(await screen.findAllByTestId('search-autocomplete-item')[0]).toBeInTheDocument();
```

Comment (evanpurkhiser) [reply static/app/views/issueList/searchBar.spec.tsx:186]: Does this work?

Comment (vuluongj20) [reply static/app/views/issueList/searchBar.spec.tsx:186]: Nope :(

Comment (evanpurkhiser) [reply static/app/views/issueList/searchBar.spec.tsx:186]: booooo

Comment (vuluongj20) [reply static/app/views/issueList/searchBar.spec.tsx:186]: For some reason it takes some time for the items to pop up

Comment (evanpurkhiser) [reply static/app/views/issueList/searchBar.spec.tsx:186]: yeah it's fine. Let's just get stuff into RTL and we can worry about performance cleanup later

File: static/app/views/issueList/searchBar.spec.tsx (hunk 4 above)
Comment (vuluongj20) [reply static/app/views/issueList/searchBar.spec.tsx]: Good idea https://github.com/getsentry/sentry/pull/40289/commits/9724d55c9338b4c5f0a58b9e47854f5b61d1d183

**Pull request merged**